homeassistant/components/synology_dsm/sensor.py

```diff
@@ -208,3 +216,42 @@ def device_info(self) -> Dict[str, any]:
             "sw_version": self._api.information.version_string,
             "via_device": (DOMAIN, self._api.information.serial),
         }
+
+
+class SynoNasSecuritySensor(SynoNasSensor):
+    """Representation a Synology Security sensor."""
+
+    @property
+    def state(self):
+        """Return the state."""
+        return getattr(self._api.security, self.sensor_type)
+
+    async def options_updated(self):
+        """Config entry options are updated, remove entity if option is disabled."""
+        await self.async_remove()
+
+    async def async_remove(self):
```

Comment:
`options_updated` and `async_remove` will be moved to `SynoNasSensor` on a next PR what will permits to select specific volume or disk we want to monitor.